opm/input/eclipse/EclipseState/Tables/TableContainer.cpp

```diff
@@ -44,77 +46,86 @@ namespace Opm {
         return result;
     }
 
-    bool TableContainer::empty() const {
+    bool TableContainer::empty() const
+    {
         return m_tables.empty();
     }
 
-
-    size_t TableContainer::size() const {
+    size_t TableContainer::size() const
+    {
         return m_tables.size();
     }
 
-
-    size_t TableContainer::max() const {
+    size_t TableContainer::max() const
+    {
         return m_maxTables;
     }
 
-    const TableContainer::TableMap& TableContainer::tables() const {
+    const TableContainer::TableMap& TableContainer::tables() const
+    {
         return m_tables;
     }
 
-
-    size_t TableContainer::hasTable(size_t tableNumber) const {
-        if (m_tables.find( tableNumber ) == m_tables.end())
-            return false;
-        else
-            return true;
+    bool TableContainer::hasTable(size_t tableNumber) const
+    {
+        return this->m_tables.find(tableNumber)
+            != this->m_tables.end();
     }
 
-
-    const SimpleTable& TableContainer::getTable(size_t tableNumber) const {
-        if (tableNumber >= m_maxTables)
+    const SimpleTable& TableContainer::getTable(size_t tableNumber) const
+    {
+        if (tableNumber >= m_maxTables) {
             throw std::invalid_argument("TableContainer - invalid tableNumber");
-
-        if (hasTable(tableNumber)) {
-            auto pair = m_tables.find( tableNumber );
-            return *(pair->second.get());
-        } else {
-            if (tableNumber > 0)
-                return getTable(tableNumber -1);
-            else
-                throw std::invalid_argument("TableContainer does not have any table in the range 0..." + std::to_string( tableNumber ));
         }
-    }
 
-
-    const SimpleTable& TableContainer::operator[](size_t tableNumber) const {
-        return getTable(tableNumber);
+        if (auto pairPos = m_tables.find(tableNumber); pairPos != m_tables.end()) {
+            return *pairPos->second;
+        }
+        else if (tableNumber > 0) {
+            return getTable(tableNumber - 1);
+        }
+        else {
+            throw std::invalid_argument {
+                "TableContainer does not have any table in "
+                "the range 0..." + std::to_string(tableNumber)
+            };
+        }
     }
 
-    void TableContainer::addTable(size_t tableNumber , std::shared_ptr<SimpleTable> table) {
-        if (tableNumber >= m_maxTables)
-            throw std::invalid_argument("TableContainer has max: " + std::to_string( m_maxTables ) + " tables. Table number: " + std::to_string( tableNumber ) + " illegal.");
+    void TableContainer::addTable(size_t tableNumber, std::shared_ptr<SimpleTable> table)
+    {
+        if (tableNumber >= m_maxTables) {
+            throw std::invalid_argument {
+                "TableContainer has max: " + std::to_string(m_maxTables) +
+                " tables. Table number: " + std::to_string(tableNumber) + " illegal."
```

Comment:
is

Reply:
> is

Good catch.  I've also switched to using `fmt::format()` to form the diagnostic message.